### PR TITLE
Countries-Webapp

### PR DESCRIPTION
**Pull request by diegoh40**
This React capstone project is about building a mobile webapp to check a list of metrics (numerical values) that I have created making use of React and Redux. I have selected an API that provides numerical data on a topic that I like and built the webapp around it.

### CODE REVIEW

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi @diegoh40, 

Good job so far! 👏 👏 
There are some issues that you still need to work on to go to the next project but you are almost there! 🏃‍♂️ 🏃‍♂️ 

## Highlights 💪 
- No linter errors
- Tests passes
- correct flow is used
- good PR

## Required Changes ♻️

- [ ] Good job so far 👏 👏  In addition to what you have done, I would suggest you implement some suggestions that I have included in the video below. In short, it's about the implementation of how data will be displayed on the `Home page` and the functionality of the `Back` button. I thought it would be much more understandable if I show it in the video. The Loom video link is here 👉 👉  [Loom video](https://www.loom.com/share/b460256db49f4bdb8952fcb55e83da24)

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by diegoh40** [comment]
the changes were implemented

image

I think probably the deployment in Netlifly is not updated yet, so I add an image of my mobile project. 
image

File: src/pages/Countries.js [hunk 1]
```diff
@@ -1,37 +1,45 @@
...
     dispatch(selectCountry(id));
   };
 
+  const worldContinents = [
+    { name: 'Africa', id: 1 },
+    { name: 'Asia', id: 2 },
+    { name: 'Europe', id: 3 },
+    { name: 'South America', id: 4 },
+    { name: 'North America', id: 5 },
+  ];
```
Comment (rustam-tolipov) [inline src/pages/Countries.js:28]: Good implementation. However, it needs some improvements. When I click once in any selection, it's not working unless I click it double. Also, sometimes it's filtering as expected, if I go to Africa it shows Asia. You can watch the video: 

https://user-images.githubusercontent.com/69011963/194684187-809557af-f1e2-4b8a-9ad6-d412b6b5c308.mov

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @diegoh40, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Great Design_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/pages/Countries.js (hunk 1 above)
Comment (diegoh40) [reply src/pages/Countries.js:28]: Hi thanks for the suggestion but I think you are seeing the old version of the app I worked on this thing you tell me about, perhaps Netlifly didn't deploy when you saw the page, please check again

please try this link https://deploy-preview-2--diegoh40-makes-great-sites.netlify.app/

Comment (rustam-tolipov) [reply src/pages/Countries.js:28]: It's possible let me check it

Comment (rustam-tolipov) [reply src/pages/Countries.js:28]: Yeah, it's working now. When I opened it showed me the thing on the video. Now you can resubmit it again. Next time your project is going to be approved.

Comment (diegoh40) [reply src/pages/Countries.js:28]: thanks Mr have a great day

**Comment by diegoh40** [comment]
The changes were made in the previous attempt, but due to the delay in deployment there was a misunderstanding, please check the conversation with the previous CR, thanks.

File: src/pages/Countries.js [hunk 2]
```diff
@@ -1,37 +1,45 @@
 import React, { useEffect } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
 import { Link } from 'react-router-dom';
-import { showCountries, selectCountry } from '../redux/Home/home';
+import { showCountries, selectCountry, showCountriesAll } from '../redux/Home/home';
```
Comment (rloterh) [inline src/pages/Countries.js:5]: - [ ] OPTIONAL
The styling of the list on your Homepage does not conform to the alternating dual-color pattern of the provided [design template](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)). Kindly ensure that each cell of the list item should have surrounding cells having **different (alternating) background color(s)**.

**Your design:**
image


**The original design:**
image

File: README.md [hunk 3]
```diff
@@ -11,7 +11,7 @@
 - Git Workflow, Git, GitHub
 
 ## Live Demo (if available)
-[Live Demo Link]()
+[Live Demo Link](https://diegoh40-makes-great-sites.netlify.app)
```
Comment (rloterh) [inline README.md:15]: - [ ] OPTIONAL
MIT license link in your ReadMe on _line 75_ is broken, kindly endeavor to fix it.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Diego,  

Your project is complete, great job! There is nothing else to say other than... it's time to merge it 🍾🍾🍾 Congratulations!  🎉💯🌟 

## Highlights 👍
- API fetch properly implemented
- Navigation working as intended
- State management with Redux properly implemented
- Good unit test coverage


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Kindly consider my comments marked **[OPTIONAL]** under this review_

Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**